### PR TITLE
Add ShaderLayer to Flutter compositor

### PR DESCRIPTION
**Pull request by abarth**
We need this to implement RenderShaderMask properly. See #1155.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @chinmaygarde

File: sky/compositor/shader_layer.cc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  ContainerLayer::Preroll(context, matrix);
+  set_paint_bounds(context->child_paint_bounds);
+}
+
+void ShaderLayer::Paint(PaintContext::ScopedFrame& frame) {
+  SkPaint paint;
+  paint.setXfermodeMode(transfer_mode_);
+  paint.setShader(shader_.get());
+  SkCanvas& canvas = frame.canvas();
+  canvas.saveLayer(&paint_bounds(), &paint);
```
Comment (chinmaygarde) [inline sky/compositor/shader_layer.cc]: Minor nit: `SkAutoCanvasRestore` with `doSave` (second argument) `false`

**Comment by chinmaygarde** [comment]
lgtm

File: sky/compositor/shader_layer.cc (hunk 1 above)
Comment (abarth) [inline sky/compositor/shader_layer.cc]: Done (also for the other layers)

**Pull request merged**